src/gentropy/method/locus_breaker_clumping.py

```diff
@@ -12,103 +11,163 @@
+                        | f.col("distance").isNull(),
+                        f.col("position"),
+                    ),
+                )
+                .withColumn(
+                    "locusStart",
+                    f.when(
+                        f.last(f.col("locusStart") - flanking_distance, True).over(
+                            w1.rowsBetween(-sys.maxsize, 0)
+                        )
+                        > 0,
+                        f.last(f.col("locusStart") - flanking_distance, True).over(
+                            w1.rowsBetween(-sys.maxsize, 0)
+                        ),
+                    ).otherwise(f.lit(0)),
+                )
+                .withColumn(
+                    "locusEnd", f.max(f.col("position") + flanking_distance).over(w2)
+                )
+                .withColumn("rank", f.rank().over(w3))
+                .filter(
+                    (f.col("rank") == 1) & (f.col("negLogPValue") > neglog_pv_cutoff)
+                )
+                .select(
+                    *columns_sumstats_columns,
+                    # To make sure that the type of locusStart and locusEnd follows schema of StudyLocus:
+                    f.col("locusStart").cast(t.IntegerType()).alias("locusStart"),
+                    f.col("locusEnd").cast(t.IntegerType()).alias("locusEnd"),
+                    f.lit(None)
+                    .cast(t.ArrayType(t.StringType()))
+                    .alias("qualityControls"),
+                    StudyLocus.assign_study_locus_id(
+                        f.col("studyId"), f.col("variantId")
+                    ).alias("studyLocusId"),
+                )
+            ),
+            _schema=StudyLocus.get_schema(),
+        )
+
+    @staticmethod
+    def _process_locus_breaker(
+        study_locus: StudyLocus,
+        sum_stats: SummaryStatistics,
+        large_loci_size: int,
+        wbc_clump_distance: int,
+        gwas_threshold: float,
+    ) -> StudyLocus:
+        """Process the locus breaker method result, and run window-based clumping on large loci.
+
+        Args:
+            study_locus (StudyLocus): StudyLocus object with locus start and end positions.
+            sum_stats (SummaryStatistics): Input summary statistics dataset.
+            large_loci_size (int): the size to define large loci which should be broken with wbc.
+            wbc_clump_distance (int): Clump distance for window-based clumping.
+            gwas_threshold (float): P-value threshold to be used in window-based clumping.
+
+        Returns:
+            StudyLocus: clumped study loci with large loci broken by window-based clumping.
+        """
+        small_loci = study_locus.filter(
+            (f.col("locusEnd") - f.col("locusStart")) <= large_loci_size
+        )
+        large_loci = study_locus.filter(
+            (f.col("locusEnd") - f.col("locusStart")) > large_loci_size
+        )
+        large_loci_ss = SummaryStatistics(
```

Comment:
There's no need to create the `large_loci_ss` temporary variable, as it is immediately passed to clumping. Instead you can say:

```python
        large_loci_wbc = (
            SummaryStatistics(
                sum_stats.df.alias("ss")
                .join(
                    large_loci.df.alias("ll"),
                    (f.col("ss.studyId") == f.col("ll.studyId"))
                    & (f.col("ss.chromosome") == f.col("ll.chromosome"))
                    & (f.col("ss.position") >= f.col("ll.locusStart"))
                    & (f.col("ss.position") <= f.col("ll.locusEnd")),
                    "inner",
                )
                .select([f.col("ss." + col) for col in sum_stats.df.columns]),
                SummaryStatistics.get_schema(),
            )
            # The resulting summary statistics already has the window-based clumping method:
            .window_based_clumping(large_loci_size, gwas_threshold)
            # Then add the locus start and end positions:
            .df.withColumns(
                {
                    "locusStart": f.col("position") - large_loci_size // 2,
                    "locusEnd": f.col("position") + large_loci_size // 2,
                }
            )
        )
```